File: web/src/api/models/ServiceConnectionResponse.ts
import type { ConnectionStatus } from './IntegrationConnection';

export interface ServiceConnectionResponse {
  id: string;
  statusCode: number;
  success: boolean;
  message: string;
  lastConnectionStatus?: ConnectionStatus;
  lastConnectionTest?: string; // ISO 8601 formatted date-time string representation of the last connection test timestamp
}

File: api/integration-execution-contract/src/main/java/com/integration/execution/contract/rest/response/ServiceConnectionResponse.java
package com.integration.execution.contract.rest.response;

import com.integration.execution.contract.model.enums.ConnectionStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ServiceConnectionResponse {
    private String id;
    private int statusCode;
    private boolean success;
    private String message;
    private ConnectionStatus lastConnectionStatus;
    private Instant lastConnectionTest;
}
